internal/plugincommon/consensus/oracle_min_observation.go

```diff
@@ -0,0 +1,68 @@
+package consensus
+
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"golang.org/x/crypto/sha3"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+type observersCounter[T any] struct {
+	data      T
+	observers map[commontypes.OracleID]struct{}
+}
+
+// OracleMinObservation provides a way to ensure a minimum number of observations for
+// some piece of data have occurred. It maintains an internal cache and provides a list
+// of valid or invalid data points.
+type OracleMinObservation[T any] interface {
```

Comment:
This API doc comment should outline why this would be used instead of `MinObservation`. I think you can say that `OracleMinObservation` ensures that duplicate observations from the same observer are only counted once? TBH this should be the default behavior, I'm not even sure we should have a separate MinObservation and OracleMinObservation.

cc @winder